README.txt
# HiveInteractiveMenu
Startup file should be PwndMonstersBotClient.py

Very little error catching in place at the moment

For the bot to run you need to install the link with proper permissions created in the discord bot configuration section.  
All names currently match my guild - you can edit if you need to.

You will need to add a new channel with a channel name matching the .env channel name attribute
Bot token and keys need to be added to the .env file.
Bot will only respond to commands in the specified channel. Type !help to see commands and !help *** command name for more details.

DEV - to get git to ignore the .env file if/when you put your keys in
Run "git update-index --skip-worktree .env"
This eliminates tracking of the env file so it will not push your keys into the repo
To undo: "git update-index --no-skip-worktree .env"

PythonApplication1/changelog.txt
2023-11-20 13:27:32.995160, ADDING BRAWL POINTS FOR BRAWL CYCLE 187
2023-11-20 13:27:33.782079, Successfully added new user Nwb071980 with initial point value 10.0
2023-11-20 13:27:33.793048, Successfully added new user .demon19. with initial point value 8.0
2023-11-20 13:27:33.802026, Successfully added new user anandkj611 with initial point value 6.0
2023-11-20 13:27:33.811997, Successfully added new user .dben with initial point value 9.0
2023-11-20 13:27:33.820974, Successfully added new user EM with initial point value 9.0
2023-11-20 13:27:33.827955, Successfully added new user genesis_0624 with initial point value 5.0
2023-11-20 13:27:33.834936, Successfully added new user valchiz with initial point value 5.0
2023-11-20 13:27:33.843913, Successfully added new user zukito with initial point value 6.0
2023-11-20 13:27:33.851892, Successfully added new user idkpdx with initial point value 7.0
2023-11-20 13:27:33.861863, Successfully added new user thedarkgane with initial point value 4.0
2023-11-20 13:27:33.870840, Successfully added new user jonnyutah52 with initial point value 4.0
2023-11-20 13:27:33.879821, Successfully added new user SLG tub3r0 with initial point value 4.0
2023-11-20 13:27:33.886798, Successfully added new user z1ddaxtobytaba with initial point value 4.0
2023-11-20 13:27:33.896771, Successfully added new user gaudge with initial point value 6.0
2023-11-20 13:27:33.905747, Successfully added new user wobs6817 with initial point value 5.0
2023-11-20 13:27:33.915721, Successfully added new user _moertoe with initial point value 6.0
2023-11-20 13:27:33.925693, Successfully added new user mxm with initial point value 6.0
2023-11-20 13:27:33.934672, Successfully added new user lordwolfen with initial point value 6.0
2023-11-20 13:28:03.749223, ADDING BRAWL POINTS FOR BRAWL CYCLE 186
2023-11-20 13:28:04.481265, Successfully added 8.0 to jonnyutah52. Old value 4.0, new value 12.0
2023-11-20 13:28:04.492232, Successfully added 8.0 to EM. Old value 9.0, new value 17.0
2023-11-20 13:28:04.502205, Successfully added 7.0 to SLG tub3r0. Old value 4.0, new value 11.0
2023-11-20 13:28:04.511182, Successfully added 6.0 to .demon19.. Old value 8.0, new value 14.0
2023-11-20 13:28:04.520159, Successfully added new user colnelangus with initial point value 6.0
2023-11-20 13:28:04.529133, Successfully added 8.0 to zukito. Old value 6.0, new value 14.0
2023-11-20 13:28:04.537112, Successfully added 5.0 to valchiz. Old value 5.0, new value 10.0
2023-11-20 13:28:04.547085, Successfully added 5.0 to genesis_0624. Old value 5.0, new value 10.0
2023-11-20 13:28:04.557059, Successfully added 8.0 to .dben. Old value 9.0, new value 17.0
2023-11-20 13:28:04.567035, Successfully added 10.0 to wobs6817. Old value 5.0, new value 15.0
2023-11-20 13:28:04.576008, Successfully added 8.0 to idkpdx. Old value 7.0, new value 15.0
2023-11-20 13:28:04.585981, Successfully added 5.0 to Nwb071980. Old value 10.0, new value 15.0
2023-11-20 13:28:04.595954, Successfully added 11.0 to _moertoe. Old value 6.0, new value 17.0
2023-11-20 13:28:04.602937, Successfully added 6.0 to gaudge. Old value 6.0, new value 12.0
2023-11-20 13:28:04.613908, Successfully added 2.0 to thedarkgane. Old value 4.0, new value 6.0
2023-11-20 13:28:04.621886, Successfully added 9.0 to lordwolfen. Old value 6.0, new value 15.0
2023-11-20 13:28:04.631858, Successfully added 6.0 to mxm. Old value 6.0, new value 12.0
2023-11-20 13:28:04.641833, Successfully added 1.0 to z1ddaxtobytaba. Old value 4.0, new value 5.0
2023-11-20 13:37:01.953046, REMOVING BRAWL POINTS FOR BRAWL CYCLE 186
2023-11-20 13:37:02.693246, Successfully added -8.0 to jonnyutah52. Old value 12.0, new value 4.0
2023-11-20 13:37:02.700227, Successfully added -8.0 to EM. Old value 17.0, new value 9.0
2023-11-20 13:37:02.711196, Successfully added -7.0 to SLG tub3r0. Old value 11.0, new value 4.0
2023-11-20 13:37:02.721174, Successfully added -6.0 to .demon19.. Old value 14.0, new value 8.0
2023-11-20 13:37:02.730152, Successfully added -6.0 to colnelangus. Old value 6.0, new value 0.0
2023-11-20 13:37:02.741117, Successfully added -8.0 to zukito. Old value 14.0, new value 6.0
2023-11-20 13:37:02.751090, Successfully added -5.0 to valchiz. Old value 10.0, new value 5.0
2023-11-20 13:37:02.761063, Successfully added -5.0 to genesis_0624. Old value 10.0, new value 5.0
2023-11-20 13:37:02.769048, Successfully added -8.0 to .dben. Old value 17.0, new value 9.0
2023-11-20 13:37:02.776023, Successfully added -10.0 to wobs6817. Old value 15.0, new value 5.0
2023-11-20 13:37:02.786993, Successfully added -8.0 to idkpdx. Old value 15.0, new value 7.0
2023-11-20 13:37:02.792977, Successfully added -5.0 to Nwb071980. Old value 15.0, new value 10.0
2023-11-20 13:37:02.799959, Successfully added -11.0 to _moertoe. Old value 17.0, new value 6.0
2023-11-20 13:37:02.809932, Successfully added -6.0 to gaudge. Old value 12.0, new value 6.0
2023-11-20 13:37:02.818911, Successfully added -2.0 to thedarkgane. Old value 6.0, new value 4.0
2023-11-20 13:37:02.828883, Successfully added -9.0 to lordwolfen. Old value 15.0, new value 6.0
2023-11-20 13:37:02.836862, Successfully added -6.0 to mxm. Old value 12.0, new value 6.0
2023-11-20 13:37:02.846834, Successfully added -1.0 to z1ddaxtobytaba. Old value 5.0, new value 4.0
2023-11-20 13:37:57.536054, ADDING BRAWL POINTS FOR BRAWL CYCLE 186
2023-11-20 13:37:58.322972, Successfully added 8.0 to jonnyutah52. Old value 4.0, new value 12.0
2023-11-20 13:37:58.332946, Successfully added 8.0 to EM. Old value 9.0, new value 17.0
2023-11-20 13:37:58.341924, Successfully added 7.0 to SLG tub3r0. Old value 4.0, new value 11.0
2023-11-20 13:37:58.351897, Successfully added 6.0 to .demon19.. Old value 8.0, new value 14.0
2023-11-20 13:37:58.363866, Successfully added 6.0 to colnelangus. Old value 0.0, new value 6.0
2023-11-20 13:37:58.366857, Successfully added 8.0 to zukito. Old value 6.0, new value 14.0

PythonApplication1/helper.py
#helper.py
#part of the PwndMondersBotClient program
#maily use to help interface between the bot and the text files
import os
import datetime
from dictionaries import ign_d
from dictionaries import d_ign
b_file = 'balances.txt'
l_file = 'changelog.txt'

#function to add points to a user
#ctx context object
#user is discord user tag '<@346523452345>'
#points is int points to be added (neg for subtract)
#memo is a note to add as a first line in the output message. if blank, ignore
def add_points(ctx, user, points, memo):
    if memo != '':
        memo = f'{memo}\n'
    if points == 0:
        return(f'{memo}Did not add or modify any points. Point value 0.')
    name = grab_user_name(ctx, user)
    if name == '-1':
        return(f'{memo}Could not locate username {user} in the guild. Nothing changed.')
    #check to see if user exists
    if user_exist(name):
        old_points=0
        with open(b_file,'r') as file:
            lines = file.readlines() # read in file lines
        newLines=[]
        for l in lines:
            if name in l:
                x = l.split(',')
                old_points = float(x[1])
                new_points = old_points+points
                newLines.append(f'{name},{new_points}\n') #overwrite correct line with new point value
            else:
                newLines.append(l)
        with open(b_file,'w') as file:
            file.writelines(newLines) #rewrite the new values and old values to file
        return (log(f'{memo}Successfully added {points} to {name}. Old value {old_points}, new value {new_points}'))
    else:
        #add user and points to end of the file
        with open(b_file,'a') as file:
            file.write(f'{name},{points}\n')
        return (log(f'{memo}Successfully added new user {name} with initial point value {points}'))
    return (log(f'{memo}Failed to add points. Should not get here. {user} {points}'))

# returns the entire balances.txt file as a string
def display_points_all():
    output = ''
    with open(b_file,'r') as f:
        for line in f.readlines():
            name = line.replace('\n','').split(',')[0]
            point = line.replace('\n','').split(',')[1]
            ign = swap_discord_name_for_ign(name)
            output = output + (f'{ign} has {point} points.\n')
        return output

# returns a specific users points value
def display_points(ctx, user):
    name = grab_user_name(ctx, user)
    with open(b_file) as f:
         for line in f.readlines():
             if name in line:
                 ign = swap_discord_name_for_ign(name)
                 point = line.replace('\n','').split(',')[1]
                 return f'{ign} has {point} points.'
    return (f'Could not find {name} in the balances file.')

#display lines number of lines from the change log
def display_log(lines):
    if lines < 100:
        with open(l_file) as f:
            last_lines = f.read().splitlines()[-lines:]
            return '\n'.join(last_lines)
    else:
        return (f'Too many lines requested. Must be under 100.')

#display lines number of lines from the change log, only for specified user
def display_log_user(ctx, user, lines):
    name = grab_user_name(ctx, user)
    if name == '-1':
        return(f'Could not locate username {user} in the guild. Nothing changed.')
    
    output = []
    if lines < 100:
        with open(l_file) as f:
            flines = f.read().splitlines()
            flines.reverse()
            for l in flines:
                if name in l:
                    output.append(l)
                    lines = lines -1
                    if lines == 0:
                        output.reverse()
                        return '\n'.join(output) #once we hit requested number of lines, return
            output.reverse()
            return '\n'.join(output)#if we get to the end, return with what we have
    else:
        return (f'Too many lines requested. Must be under 100.')

#helper function to record to log file and return same string recorded to output
def log(msg):
    now = datetime.datetime.now()
    with open(l_file, 'a') as f:
        f.write(f'{now}, {msg}\n')
    return msg


#helper function to check and see if a user exists in the Balances file
#user is user NAME not the discord tag number
def user_exist(user):
    with open(b_file) as f:
        if user in f.read():
            return True # found the name, it exists
    return False # name does not exist

#helper function to grab the user's NAME from the context using the discord tag
def grab_user_name(ctx, user):
    for m in ctx.guild.members:
        if str(m.id) in user:
            return m.name
    return '-1'

#helper function to switch from the discord name to the in game name
#if it cannot find the name in the dictionary, just returns the discord name back
def swap_discord_name_for_ign(discord_name):
    ign = d_ign[discord_name]
    if ign ==  '':
        ign = discord_name #if we cant find in game name, replace with name
    return ign

#helper function to switch from the in game name to the discord PING name 
#if it cannot find the name in the dictionary, does nothing
def swap_ign_for_discord_ping(ign):
    ping = ign_d[ign]
    if ping ==  '':
        ping = ign #if we cant find it, do nothing
    return ping

PythonApplication1/SPLhelper.py
import json
import requests
import random
import string
import os
import discord
from beem.transactionbuilder import TransactionBuilder
from beembase.operations import Custom_json
from helper import *
from dictionaries import ign_d
from dictionaries import d_ign
from dictionaries import tier3_bonus
from dictionaries import tier4_bonus

pwndID = 'c43af991dc15970e2047548f5b7bfc30a9943543'
#this file is to put Splinterlands API calls and interactions here

#Add Brawl Data To Users
#cycle should be cycle number,
#toAdd is BOOL whether to Add (true) or Subtract (false) scores
def recordBrawl(ctx, cycle, toAdd):
    #first step - identify the tournament ID based on the cycle number and guild ID
    t_ID = ''
    try:
        r = requests.get(f'https://api.splinterlands.com/guilds/brawl_records?guild_id={pwndID}&cycle={cycle}')
        jsonData = json.loads(r.text)
        t_ID = jsonData['results'][0]['tournament_id']
    except Exception as e:
        return f'Error trying to find the Tournament ID from guilds/brawl_records! \n {e}'

    output = []
    #second step - pull json brawl data
    try:
        r = requests.get(f'https://api.splinterlands.com/tournaments/find_brawl?id={t_ID}&guild_id={pwndID}')
        jsonData = json.loads(r.text)
        #iterate through players and record results
        for p in jsonData['players']:
            battles_entered = p['entered_battles']
            wins = p['wins']
            losses = p['losses']
            draws = p['draws']
            name = p['player']
            fray = p['fray_index']
            brawl = p['brawl_level']
            output.append(calculatePoints(ctx, name, battles_entered, wins, losses, draws, fray, brawl, cycle, toAdd))
    except Exception as e:
        return f'Error trying to find the brawl data from /tournaments/find_brawl records! \n {e}'

    return '\n'.join(output) #returns the messages back, one line per user

def calculatePoints(ctx, name, battles_entered, wins, losses, draws, fray, brawl, cycle, toAdd):
    memo = name
    if battles_entered == 0:
        idk = ign_d['idkpdx']
        return f'User {ign_d[name]} did NOT enter battles for brawl cycle {cycle}. {idk}'
    total = 0
    #current algorithm = 1 point per win and half point for draws
    total = total + wins
    memo = f'{memo} +{wins} for wins,'
    if draws > 0:
        total = total + (0.5*draws)
        memo = f'{memo} +{draws} for draws (half points),'
    
    #3 bonus points for 0 losses, 1 bonus point for 1 loss
    if losses == 0:
        total = total + 3
        memo = f'{memo} +3 for no losses,'
    elif losses == 1:
        total = total + 1
        memo = f'{memo} +1 for only 1 loss,'
    #bonus points for tougher fray assignemnts
    fray_bonus = 0
    if brawl == 6:
        fray_bonus = tier3_bonus[str(fray)]        
    elif brawl == 8:
        #TODO need to check this!! not sure this is corrrect
        fray_bonus = tier4_bonus[str(fray)]
    else:
        #todo??
        return f'HELP not sure what to do here, unexpected brawl value in calculatePoints {brawl}'
    total = total + fray_bonus
    memo = f'{memo} and +{fray_bonus} for fray difficulty. +{total} total.'
    if toAdd == False:
        total = total * -1 #if needed to delete points from a brawl added erroniously or doubled then do so here
    else:
        y = 3
        #message_splicer_sender(ctx,memo) #send log line to illustrate how points were calculated
    return(add_points(ctx,ign_d[name],total,memo))

PythonApplication1/PwndMonstersBotClient.py

# bot.py
import os
import random
import discord
from helper import *
from SPLhelper import *
from dotenv import load_dotenv
from discord.ext import commands

load_dotenv()
TOKEN = os.getenv('DISCORD_TOKEN') #app token
CHANNEL = os.getenv('CHANNEL_NAME') #channel name to interact with the bot
R_ACTIVE = os.getenv('REWARD_ACCOUNT_ACTIVE_KEY') #reward account active key for auto-reward distribution
R_POSTING = os.getenv('REWARD_ACCOUNT_POSTING_KEY') #reward accont posting key for auto-reward distribution

intents = discord.Intents.default()
intents.members = True
bot = commands.Bot(command_prefix='!',intents = intents)

@bot.event
async def on_ready(): #this method happens upon initialization of the bot
    print(f'{bot.user.name} has connected to Discord!\n'
            f'{bot.user.name} is connected to the following guild(s):\n')
    for g in bot.guilds:
        print(f'{g.name}(id: {g.id})')
        members = '\n'
        for member in g.members:
            members =  members + '-' + member.name + '\n'
        print(f'Guild Members: {members}')
    print('Initialized - awaiting events...')

@bot.event
async def on_message(message): #occurs when a new message is typed
    if message.author == bot.user:
        return #if bot is seeing its own message, ignore
    if message.channel.name != CHANNEL:
        return #if message is NOT in "pwnd-bot" channel, ignore
    if message.content == 'ping':
        await message.channel.send('pong - on_message')
    await bot.process_commands(message) #sends commands to the commands methods after through this one

@bot.command(name="add_Brawl", help='*** {cycle} - Add the brawl results and tally points for given brawl cycle. (***Officers only)')
@commands.has_role('Officers')
async def add_Brawl(ctx, cycle: int):
    log(f'ADDING BRAWL POINTS FOR BRAWL CYCLE {cycle}')
    await message_splicer_sender(ctx, recordBrawl(ctx, cycle, True)) #function, should return string
    # await ctx.send(recordBrawl(ctx, cycle, True)) 

@bot.command(name="remove_Brawl", help='*** {cycle} - Subtract the brawl results and tally points for given brawl cycle.  (***Officers only)')
@commands.has_role('Officers')
async def remove_Brawl(ctx, cycle: int):
    log(f'REMOVING BRAWL POINTS FOR BRAWL CYCLE {cycle}')
    await message_splicer_sender(ctx, recordBrawl(ctx, cycle, False)) #function, should return string
    # await ctx.send(recordBrawl(ctx, cycle, False)) 

@bot.command(name="change_Log", help='*** {} - Show the last 99 lines from the changelog. (***Officers only)')
@commands.has_role('Officers')
async def change_Log(ctx):
    await message_splicer_sender(ctx,display_log(99)) # calls helper function, should return string
    #await ctx.send(display_log(99)) 

@bot.command(name="change_Log_N", help='*** {num} - Show the last N lines from the changelog. (***Officers only)')
@commands.has_role('Officers')
async def change_Log_N(ctx, lines: int):
    await message_splicer_sender(ctx, display_log(lines)) # calls helper function, should return string
    #await ctx.send(display_log(lines)) 

@bot.command(name="change_Log_Name", help='*** {@name} - Show the last 99 lines from the changelog for specific user. (***Officers only)')
@commands.has_role('Officers')
async def change_Log_Name(ctx, user: str):
    await message_splicer_sender(ctx,display_log_user(ctx, user, 99)) # calls helper function, should return string

@bot.command(name="change_Log_Name_N", help='*** {@name, num} - Show the last N lines from the changelog for specific user. (***Officers only)')
@commands.has_role('Officers')
async def change_Log_Name_N(ctx, user: str, lines: int):
    await message_splicer_sender(ctx,display_log_user(ctx, user, lines)) # calls helper function, should return string
    #await ctx.send(display_log_user(ctx, user, lines)) 

@bot.command(name="add_Points", help='*** {@name, num} - Add or subtract (with negative value) point value from a specific user. (***Officers only)')
@commands.has_role('Officers')
async def add_Points(ctx,user: str, points: int):
    # this one should always only be 1 line, dont need splicer method here
    await ctx.send(add_points(ctx, user, points,'')) # calls helper function, should return string

@bot.command(name="ping", help='{} Heartbeat to verify if the bot is running correctly.')
async def ping(ctx):
    #user = '<@876552577079197727>'
    #await ctx.send(f'pong for {user}')
    #await ctx.send('pong for <@1175587765203767317>')
    await ctx.send('pong')

@bot.command(name='display_All', help='{} Displays the current points table.')
async def display_all(ctx):
    await message_splicer_sender(ctx,display_points_all())
    #await ctx.send(display_points_all())

@bot.command(name='display_Name', help='{@name} Displays a specific user\'s current points.')
async def display_name(ctx, user):
    # this one should always only be 1 line, dont need splicer method here
    await ctx.send(display_points(ctx, user))

#split lines of a message up into chunks of 18 lines each.
async def message_splicer_sender(ctx, msg):
    lines = msg.split('\n')
    limit = 18
    if len(lines) < limit:
        await ctx.send(msg)
    else:
        #split into multiple messages
        m = []
        x = 0
        for i in range(0, len(lines)):
            m.append(lines[i])
            x = x+1
            if x == limit:
                x=0
                await ctx.send('\n'.join(m))
                m = []
        if m != []:
            await ctx.send('\n'.join(m)) #send leftovers if they exist

#error handling
@bot.event
async def on_error(event, *args, **kwards):
    with open('err.log', 'a') as f:
        if event == 'on_message':
            f.write(f'Unhandled message: {args[0]}\n')
        else: 
            raise

@bot.event
async def on_command_error(ctx, error):
    if isinstance(error, commands.errors.CheckFailure):
        await ctx.send('Only Officers may run this command.')

#testings methods, don't need anymore
#@bot.command(name='roll_dice', help='Simulate a dice roll.')
#@commands.has_role('Officers')
#async def roll_dice(ctx, number_of_dice: int, number_of_sides: int):
#    dice = [ str(random.choice(range(1, number_of_sides +1)))
#            for _ in range(number_of_dice)]
#    await ctx.send(', '.join(dice))

bot.run(TOKEN)

PythonApplication1/PwndMonstersBotClient_backup.py

# bot.py
import os
import random
import discord
from dotenv import load_dotenv
from discord.ext import commands

load_dotenv()
TOKEN = os.getenv('DISCORD_TOKEN')
GUILD = os.getenv('DISCORD_GUILD')

intents = discord.Intents.default()
intents.members = True

client = discord.Client(intents = intents)


@client.event
async def on_ready():
    guild = discord.utils.get(client.guilds, name=GUILD)
    #guild = discord.utils.find(lambda g: g.name == GUILD, client.guilds)
    #for guild in client.guilds:
    #    if guild.name == GUILD:
    #        break

    print(
        f'{client.user} has connected to Discord!\n'
        f'{client.user} is connected to the following guild:\n'
        f'{guild.name}(id: {guild.id})'
        )

    members = ''
    for member in guild.members:
        members =  members + '\n -' + member.name

    
    print(f'Guild Members:\n - {members}')
    print('Initialized - awaiting events...')

@client.event
async def on_message(message):
    if message.author == client.user:
        return #if bot is seeing its own message, ignore
    
    if message.channel.name != 'pwnd-bot':
        return #if message is NOT in "pwnd-bot" channel, ignore

    #officer powers

    #everyone powers
    
    if message.content == 'ping':
        await message.channel.send('pong')

    brooklyn_99_quotes = [
        'I\'m the human form of the 💯 emoji.',
        'Bingpot!',
        (
            'Cool. Cool cool cool cool cool cool cool, '
            'no doubt no doubt no doubt no doubt.'
        ),
    ]

    if message.content== '99!':
        response = random.choice(brooklyn_99_quotes)
        await message.channel.send(response)
    elif message.content == 'raise-exception':
        raise discord.DiscordException

@client.event
async def on_error(event, *args, **kwards):
    with open('err.log', 'a') as f:
        if event == 'on_message':
            f.write(f'Unhandled message: {args[0]}\n')
        else: 
            raise

client.run(TOKEN)

PythonApplication1/dictionaries.py
#dictionary to go from in game name to PING name
ign_d = {
    'demon19' : '<@911775022169858049>',
    'wobs' : '<@876552577079197727>',
    'idkpdx' : '<@701899884508217356>',
    'moertoe' : '<@728568906834378793>',
    'jonny-utah' : '<@1032453361351524432>',
    'playtoearn1992' : '<@891541607109771314>',
    'tub3r0' : '<@868080832647135242>',
    'chichie0214' : '<@876261466355925013>',
    'valchiz' : '<@343112576974061569>',
    'thedarkgane' : '<@544976139110842378>',
    'nwb0719' : '<@949193580067553321>',
    'z1ddax' : '<@804048821394997279>',
    'plebdev' : '<@200427154247254025>',
    'zukito' : '<@932391846343225364>',
    'elrmapa' : '<@816235923330564107>',
    'gaudgeo' : '<@953806899923521646>',
    'mxm0unite' : '<@855079086291288064>',
    'lord-wolfen' : '<@234915223281205250>',
    'contactmike1' : '<@717048190645436466>',
    'sofia06' : '<@972041408405995520>',
    'luckygamer' : '<@280326226474958849>',
    'magus-tenebrae' : '<@958779047435960330>',
    'datafreak' : '<@728568906834378793>',
    'hajime001' : '<@809287895969169408>',
    'colnelangus' : '<@463907572806713345>',
    'thebusycat' : '<@745051200436699177>'
    }

#dictionary to go from discord name to in game name
d_ign = {
    'Nwb071980' : 'nwb0719' ,
    '.demon19.' : 'demon19' ,
    'anandkj611' : 'playtoearn1992' ,
    '.dben' : 'plebdev' ,
    'EM' : 'elrmapa' ,
    'genesis_0624' : 'chichie0214' ,
    'valchiz' : 'valchiz' ,
    'zukito' : 'zukito' ,
    'idkpdx' : 'idkpdx' ,
    'thedarkgane' : 'thedarkgane' ,
    'jonnyutah52' : 'jonny-utah',
    'SLG tub3r0' : 'tub3r0' ,
    'z1ddaxtobytaba' : 'z1ddax' ,
    'gaudge' : 'gaudgeo' ,
    'wobs6817' : 'wobs' ,
    '_moertoe' : 'moertoe' ,
    'mxm' : 'mxm0unite' ,
    'lordwolfen' : 'lord-wolfen',
    'colnelangus' : 'colnelangus' ,
    'contactmike1' : 'contactmike1' ,
    'mephisto_angel' : 'luckygamer' ,
    'hajime_001' : 'hajime001' ,
    'magustenebrae' : 'magus-tenebrae' ,
    'colnelangus' : 'colnelangus' ,
    'TBC' : 'thebusycat' ,
    '.miks12' : 'sofia06'
    }

#remember!! Frays are indexed at 0! so Fray 1 in SPL is 0 here
tier3_bonus = {
    '0':0,
    '1':0,
    '2':0,
    '3':0,
    '4':0,
    '5':3,
    '6':0,
    '7':0,
    '8':0,
    '9':0,
    '10':5,
    '11':1,
    '12':1,
    '13':1,
    '14':3,
    '15':3,
    '16':5,
    '17':5
    }

#remember!! Frays are indexed at 0! so Fray 1 in SPL is 0 here
tier4_bonus = {
    '0':0,
    '1':0,
    '2':0,
    '3':0,
    '4':0,
    '5':0,
    '6':0,
    '7':0,
    '8':0,
    '9':5,
    '10':1,
    '11':1,
    '12':1,
    '13':1,
    '14':7,
    '15':3,
    '16':3,
    '17':3,
    '18':5,
    '19':5,
    '20':5
    }
